mobify/platform-scaffold
# Updated README and added new cert for Windows users

## Progressive Web Scaffold - Hotfix release!
- [x] Branch off of `master` and create a `hotfix-documentation-windows-install` branch.
- [x] Create a new pull request with the following settings:
      * Base: `master`
      * Compare: `hotfix-documentation-windows-install`
      Paste the contents of this checklist into this pull request.
- [x] Update the version of the `progressive-web-sdk` npm module in `package.json`.
- [x] Make necessary changes to make it compatible with the latest release of the `progressive-web-sdk`.
- [x] In the CHANGELOG.md file of this release branch, change the 'To be released' header to 'vX.Y.Z'.
- [ ] Merge `release-vX.Y.Z` into `master` using the pull request and delete the release branch.
- [ ] Draft a new Github release with the following settings:
      * Tag version: `X.Y.Z` @ `master`
      * Release title: `X.Y.Z - <release_name>`
      * Description: Use highlights from the CHANGELOG.md (only pick out the most significant changes)
- [ ] Merge `master` into `develop` (no need for review on PR, just merge).

## Bundle Release
- [ ] Run `npm run push -- -m '<X.Y.Z>:<COMMIT_SHA>,<BRANCH_RELEASED_FROM>'`. You can find your `<COMMIT_SHA>` from `git rev-parse --short HEAD`

# Changes

On Windows 10 (not sure about earlier versions), it is not possible to add a `.pem` file to the Windows Trusted Root Certificate Store to suppress SSL errors during development. They will need a `.p7b` file instead. To support our Windows users on upcoming training events, we wanted to create the `.p7b` file for them and include it in the scaffold. 

**Security note:** I converted our `.pem` file `.p7b` format using a third-party site called [SSLShopper](https://www.sslshopper.com/). If this is not kosher from a security standpoint (do we fully trust the converted file?), please say so!

I've also added clarification about having to run npm scripts from the web directory inside the project directory since our users frequently get tripped up by this during training.

**Question**: Should we be describing these as "HTTPS errors" (as we do in the section title) or "security warnings" (as we do in the body text)? If we change the section title to match, we may have to update deep links to this section in a few places.

## Changes
- Added `.p7b` file
- Added new section at the top of README: Running npm scripts
- Edited instructions for HTTPS errors for macOS and Windows SSL for accuracy and clarity

## How to test-drive this PR
- Open README.md
- Follow instructions for "Avoiding HTTPS errors in local development" 
- Follow both macOS and Windows instructions on fresh (virtual) machines, if possible
- Verify that both sets of steps are numbered properly
- Verify that step #4 in the macOS instructions makes sense (I don't have a fresh config to test)
- Make sure to add the `.p7b` file when following the Windows instructions
- Verify that all you have to is close the browser and reopen it for the new certificate to take effect. Do you also have to clear Mobify cookies? Is this behaviour the same for Firefox (and others) as it is with Chrome?
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- There should be no security warnings or errors, as we promise
